python/pyspark/ml/param/shared.py

```diff
@@ -311,7 +311,7 @@ class HasSeed(Params):
 
     def __init__(self):
         super(HasSeed, self).__init__()
-        self._setDefault(seed=hash(type(self).__name__))
+        self._setDefault(seed=None)
```

Comment:
I could be wrong, but I think if the `Param` var is a `Long` on the Scala side, Py4j will convert a `None` to 0.  Does this sound right @holdenk ?